### DISCUSSION TITLE
Can anyone explain the version number discrepancy between the recent builds in Github actions and the contents of the snapcraft.io?

### DISCUSSION THREAD

**Discussion by ajkuck** [VSCodium/vscodium#1787] [Q&A] [answered]
This troubling me, but it isn't a bug, so I figured I'd ask here.  

At the time I'm writing this I've noticed that there was a build of stable-linux on Jan 19 for version 1.85.2.2401. (https://github.com/VSCodium/vscodium/actions/runs/7587594236/job/20668898007)  That build apparently uploaded to the snapcraft repository successfully according to the logs.

Right now, [https://snapcraft.io/codium] is showing the most recent release as version 1.84.2.23317.    That version is maybe from the release performed on Nov 13, 2023? (I had a hard time find a corresponding build output in the github actions logs.)

My initial guess was that there was some sort of delay between when a package was uploaded and when it was made available for download.  However...

The raw logs for that job show that the github action is checking what's on snapcraft and comparing to github.
```
2024-01-19T18:34:28.1402802Z Snap version: 1.84.2.23319
2024-01-19T18:34:30.0378555Z GH version: 1.85.2.24019
```
Note the snapcraft build number is 23319 not 23317.  

To deepen the mystery, this all came to my attention because one of my machines updated it's version of codium to version 23317 recently.  However, another machine that had not yet updated shows this snap info for codium:
```
$ snap info codium
name:      codium
summary:   Code editing. Redefined.
publisher: VSCodium
store-url: https://snapcraft.io/codium
contact:   https://github.com/VSCodium/vscodium/issues
license:   unset
description: |
  Binary releases of Code without branding/telemetry/licensing
commands:
  - codium
  - codium.url-handler
snap-id:      lIZWXTqmo6LFSts5Cgk2VPlNwtysZAeH
tracking:     latest/stable
refresh-date: 2023-11-21
channels:
  latest/stable:    1.84.2.23317 2024-01-19 (362) 183MB classic
  latest/candidate: ↑                                   
  latest/beta:      ↑                                   
  latest/edge:      ↑                                   
installed:          1.84.2.23319            (365) 183MB classic
```

So now I'm looking at three distinct versions of codium.
* version 1.85.2.2401 (the latest)
   * The github action logs claim it was built and uploaded to snapcraft successfully on Jan 19th 2024.
* version 1.84.2.23319 
   * Installed on my machine back in November
   * The github-action logs observed this as the version available on snapcraft at the start of the build on the 19th
* version 1.84.2.23317
   * The version snapcraft declares as having been made available to the snapcraft store on Jan 19th.

How is it that the version I installed 2 months ago is somehow newer than the one published 5 days ago?

Did some sort of roll-back happen?  Is there a bug in the github-action workflow somewhere?  Does snapcraft actually have any sort of traceability somewhere that might reveal this?

If anyone happens to know, I'm curious what could be going on.

**Comment by daiyam** [comment, marked as answer]
The `1.85` versions are incompatible with the `core18` used to build the snap. So I've reverted the snap to a working version (when the `1.85` was published).
I should fix those compatibility issues soon.

**Reply by ajkuck** [reply]
I appreciate the quick response!   

That kind of makes sense.  If the incompatibility showed up under version 1.85.x, I'm still a little unclear on why it was reverted to version 1.84.2.23317 when 1.84.2.23319 was already published.

Also - for future reference - is there a away to tell where that reversion occurred.  Or was it a manual process outside of any of the github-action logs?

**Reply by daiyam** [reply]
> why it was reverted to version 1.84.2.23317 when 1.84.2.23319 was already published.

My bad, I did make a mistake. I've set the stable channel to `1.84.2.23319`.
